File: bot-telegram/set_webhook.py
#!/usr/bin/env python3
"""
Configura il webhook Telegram per puntare a Vercel
"""
import sys
import io
if sys.platform == 'win32':
    sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')
import requests
import os

# Token bot
BOT_TOKEN = "8172610054:AAH4RDMjU3GkKvYil27hBKSiUJMi7SmMa8U"

# URL Vercel production
VERCEL_URL = "https://aiforge-pro-mattiadeblasio94-8016s-projects.vercel.app"
WEBHOOK_URL = f"{VERCEL_URL}/api/telegram/webhook"

# Configura webhook
res = requests.get(f"https://api.telegram.org/bot{BOT_TOKEN}/setWebhook?url={WEBHOOK_URL}")
data = res.json()

if data.get("ok"):
    print(f"✅ Webhook configurato: {WEBHOOK_URL}")
    print(f"Info: {data.get('result', {})}")
else:
    print(f"❌ Errore: {data}")

File: pages/api/telegram/set-webhook.js
/**
 * TELEGRAM WEBHOOK SETUP API
 * Configura il webhook di Telegram per puntare a questo endpoint
 */

export const config = {
  api: {
    bodyParser: true,
  },
}

export default async function handler(req, res) {
  // Bypass Vercel bot protection
  res.setHeader('x-vercel-protection-bypass', 'telegram-webhook-setup')

  // Proteggi endpoint - solo admin
  const { secret } = req.query;
  if (secret !== process.env.WEBHOOK_SECRET) {
    return res.status(403).json({ error: 'Accesso negato' });
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Metodo non consentito' });
  }

  const BOT_TOKEN = process.env.TELEGRAM_BOT_TOKEN;
  const APP_URL = process.env.NEXT_PUBLIC_APP_URL || 'https://aethersy.com';

  if (!BOT_TOKEN) {
    return res.status(500).json({
      error: 'TELEGRAM_BOT_TOKEN non configurato nelle variabili d\'ambiente Vercel'
    });
  }

  const webhookUrl = `${APP_URL}/api/telegram/webhook`;

  try {
    // Imposta webhook
    const setResponse = await fetch(
      `https://api.telegram.org/bot${BOT_TOKEN}/setWebhook?url=${encodeURIComponent(webhookUrl)}&allowed_updates=update_id,message,callback_query`,
      { method: 'GET' }
    );

    const setResult = await setResponse.json();

    if (!setResult.ok) {
      return res.status(500).json({
        success: false,
        error: setResult.description || 'Errore impostazione webhook'
      });
    }

    // Verifica webhook
    const infoResponse = await fetch(
      `https://api.telegram.org/bot${BOT_TOKEN}/getWebhookInfo`,
      { method: 'GET' }
    );

    const infoResult = await infoResponse.json();

    return res.json({
      success: true,
      message: 'Webhook configurato con successo',
      webhookUrl,
      info: infoResult.result
    });

  } catch (error) {
    console.error('Set webhook error:', error);
    return res.status(500).json({
      success: false,
      error: error.message
    });
  }
}
